From d238a729c67450b74b854931d0c1bb6174447faa Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Sat, 7 Aug 2021 11:11:39 +0200
Subject: [PATCH] When aux cameras are enabled, ignore "system only" camera
 flag (it shouldnt be q security issue since secure cameras are listed
 otherwise)

---
 .../libcameraservice/common/CameraProviderManager.cpp  | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/services/camera/libcameraservice/common/CameraProviderManager.cpp b/services/camera/libcameraservice/common/CameraProviderManager.cpp
index d382cf49bb..888b76346c 100644
--- a/services/camera/libcameraservice/common/CameraProviderManager.cpp
+++ b/services/camera/libcameraservice/common/CameraProviderManager.cpp
@@ -991,6 +991,16 @@ SystemCameraKind CameraProviderManager::ProviderInfo::DeviceInfo3::getSystemCame
             entryCap.data.u8[0] == ANDROID_REQUEST_AVAILABLE_CAPABILITIES_SECURE_IMAGE_DATA) {
         return SystemCameraKind::HIDDEN_SECURE_CAMERA;
     }
+
+    if(!property_get_bool("persist.sys.phh.include_all_cameras", false)) {
+        for (size_t i = 0; i < entryCap.count; ++i) {
+            uint8_t capability = entryCap.data.u8[i];
+            if (capability == ANDROID_REQUEST_AVAILABLE_CAPABILITIES_SYSTEM_CAMERA) {
+                return SystemCameraKind::SYSTEM_ONLY_CAMERA;
+            }
+        }
+    }
+
     return SystemCameraKind::PUBLIC;
 }
 
-- 
2.25.1

